compiler/rustc_hir_typeck/src/pat.rs

```diff
@@ -1531,9 +1531,12 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
             .filter(|(_, ident)| !used_fields.contains_key(ident))
             .collect::<Vec<_>>();
 
-        let inexistent_fields_err = if !(inexistent_fields.is_empty() || variant.is_recovered())
+        let inexistent_fields_err = if !inexistent_fields.is_empty()
             && !inexistent_fields.iter().any(|field| field.ident.name == kw::Underscore)
         {
+            if let Some(guard) = variant.has_errors() {
```

Comment:
Make `has_errors` into a `Result<(), ErrorGuaranteed>` so this can just be `variant.has_errors()?;`